tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -0,0 +1,73 @@
+package thredds.server.wfs;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import thredds.server.wfs.WFSRequestType;
+
+/**
+ * Controller for WFS Simple Geometry Web Service
+ * 
+ * @author wchen@usgs.gov
+ *
+ */
+@Controller
+@RequestMapping("/wfs")
+public class WFSController extends HttpServlet {
+
+
+	/**
+	 * Processes GetCapabilities requests.
+	 * 
+	 * @param out
+	 * @return
+	 */
+	private void getCapabilities(PrintWriter out, HttpServletRequest hsreq) {
+		WFSGetCapabilitiesWriter gcdw = new WFSGetCapabilitiesWriter(out);
+		gcdw.startXML();
+		gcdw.setServer(hsreq.getScheme() + "://" + hsreq.getServerName() + ":" + hsreq.getServerPort() + "/thredds/wfs");
+		gcdw.addOperation(WFSRequestType.GetCapabilities); gcdw.addOperation(WFSRequestType.DescribeFeatureType); gcdw.addOperation(WFSRequestType.GetFeature);
+		gcdw.writeOperations();
+		gcdw.addFeature(new WFSFeature("hru_soil_moist", "HRU Soil Moisture"));
+		gcdw.writeFeatureTypes();
+		gcdw.finishXML();
+	}
+	
+	/**
+	 * A handler for WFS based HTTP requests that sends to other request handlers
+	 * to handle the request.
+	 *
+	 * Servlet Path: /wfs/{request}
+	 *
+	 */
+	@RequestMapping("**")
+	public void httpHandler(HttpServletRequest hsreq, HttpServletResponse hsres) {
+		try {
+			PrintWriter wr = hsres.getWriter();
+			String request = hsreq.getParameter("REQUEST");
+			
+			// 
+			if(request != null) {
+				if(request.equals(WFSRequestType.GetCapabilities.toString())) {
+					getCapabilities(wr, hsreq);
+				}
+			}
+			
+			else{

```

Comment:
Agreed, yeah the default message was kind of a funny little thing will be better to make it that HTTP 404 (though wfs servers seem to implement this ows:Exception i.e. http://demo.mapserver.org/cgi-bin/wfs?VERSION=2.0.0&SERVICE=wfs so should 404 be wrapper inside one of these?)!

Reply:
Okay so this will be handled using the ows:Exception according to the standard. However, when GetFeature is implemented, using HTTP404 to handle non-existing datasets seems like the way to handle those things.

Reply:
Yesss! Good call! I had forgotten about the tendency to return Exception messages in XML.